librad/src/meta/entity.rs

```diff
@@ -205,9 +254,11 @@ pub trait Resolver<T> {
 ///   and certifiers, to prove that the entity evolution is actually under the
 ///   control of its current "owners" (the idea is taken from [TUF](https://theupdateframework.io/)).
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub struct Entity<T> {
+pub struct Entity<T, ST> {
     /// Entity verification status
     status: VerificationStatus,
+    /// Verification status marker type
+    status_marker: ST,
```

Comment:
So I was expecting to see this as `_status_marker: PhantomData<ST>`. The above status types would then be types that don't have constructors, i.e.:
```rust
pub struct Unknown;
pub struct Signed;
pub struct Verified;
```

Then the return type of `check_signatures` for `Entity` would be: `Entity<T, Signed>`

Does this make sense to you? Maybe this wouldn't work for some reason, but I can't tell without following the types myself.